testproject/Assets/Tests/Manual/NetworkAnimatorTests/AnimatedCubeController.cs

```diff
@@ -13,32 +13,34 @@ public class AnimatedCubeController : NetworkBehaviour
         private bool m_Rotate;
         private NetworkAnimator m_NetworkAnimator;
         private bool m_IsServerAuthoritative = true;
+        public bool IsServerAuthoritative => m_IsServerAuthoritative;
 
-        private void Awake()
+        public override void OnNetworkSpawn()
         {
+            if (HasAuthority())
+            {
+                enabled = false;
+            }
             m_Animator = GetComponent<Animator>();
-            m_NetworkAnimator = GetComponent<NetworkAnimator>();
+            m_NetworkAnimator = GetComponent<OwnerNetworkAnimator>();
             if (m_NetworkAnimator == null)
             {
-                m_NetworkAnimator = GetComponent<OwnerNetworkAnimator>();
+                m_NetworkAnimator = GetComponent<NetworkAnimator>();
                 if (m_NetworkAnimator != null)
                 {
-                    m_IsServerAuthoritative = false;
+                    m_IsServerAuthoritative = true;
```

Comment:
Can we have `m_IsServerAuthoritative = (m_NetworkAnimator != null);` outside the if, instead? As-is, it gives the impression that if `m_NetworkAnimator` is not null, `m_IsServerAuthoritative` keeps it previous value, which forces one to then wonder "is it a default false, or does it persist previous value?"